### DISCUSSION TITLE
git add only modified changes and ignore untracked files

### DISCUSSION THREAD

**Discussion by lucky18210** [community/community#29316] [Other Feature Feedback, Questions, & Ideas] [answered]
I ran "git status" and listed below are some files that were modified/or under the heading "changes not staged for commit". It also listed some untracked files that I want to ignore (I have a ".gitignore" file in these directories).

I want to put the modified files in staging so I can commit them. When I ran "git add .", it added the modified files AND the files I want to ignore to staging.

How do I add only the modified files and ignore the untracked files if presented with the git status below.

Also, are my ".gitignore" files working properly?

**Comment by luck18210** [comment, marked as answer]
Ideally your **_.gitignore_** should prevent the untracked (and ignored) files from being shown in status, added using **_git add_** etc. So I would ask you to correct your **_.gitignore_**

You can do **_git add -u_** -u so that it will stage the modified and deleted files.

You can also do **_git commit -a_** to commit only the modified and deleted files.

Note that if you have Git of version before 2.0 and used git add ., then you would need to use **_git add -u_** . (See "Difference of “git add -A” and “git add .”").

**Reply by lucky18210** [reply]
Thank you for your reply.
Where can I reference the Difference of “git add -A” and “git add?”

**Reply by luck18210** [reply]
Sorry,
Please refer the following url.
https://devilswroclaw.pl/host-https-stackoverflow.com/a/16162511/6309

**Reply by lucky18210** [reply]
Ty
